iap/generate_self_signed_jwt.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2024 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import datetime
+import json
+
+import google.auth
+from google.cloud import iam_credentials_v1
+import jwt
+
+
+def generate_jwt_payload(service_account_email: str, resource_url: str) -> str:
+    """Generates JWT payload for service account.
+
+    The resource url provided must be the same as the url of the IAP secured resource.
+
+    Args:
+      service_account_email (str): Specifies service account JWT is created for.
+      resource_url (str): Specifies scope of the JWT, the URL that the JWT will be allowed to access.
+    Returns:
+      A signed-jwt that can be used to access IAP protected applications.
+        Access the application with the JWT in the Authorization Header.
+        curl --verbose --header 'Authorization: Bearer SIGNED_JWT' URL
```

Comment:
Be consistent with formatting:
```suggestion
      Access the application with the JWT in the Authorization Header.
      curl --verbose --header 'Authorization: Bearer SIGNED_JWT' URL
```